crates/bevy_reflect/src/impls/std.rs

```diff
@@ -213,6 +213,22 @@ macro_rules! impl_reflect_for_veclike {
         }
 
         impl<T: FromReflect> List for $ty {
+            fn insert(&mut self, index: usize, value: Box<dyn Reflect>) {
+                let value = value.take::<T>().unwrap_or_else(|value| {
+                    T::from_reflect(&*value).unwrap_or_else(|| {
+                        panic!(
+                            "Attempted to insert invalid value of type {}.",
+                            value.type_name()
+                        )
+                    })
+                });
+                $insert(self, index, value);
+            }
+
+            fn remove(&mut self, index: usize) -> Box<dyn Reflect> {
+                Box::new($remove(self, index)) as Box<dyn Reflect>
```

Comment:
Nit: I know the other methods already have it, but I don't think this cast is necessary, right?
```suggestion
                Box::new($remove(self, index))
```